internal/ethereum/deploy_contract_prepare.go

```diff
@@ -95,7 +96,9 @@ func (c *ethConnector) prepareDeployData(ctx context.Context, req *ffcapi.Contra
 	ethParams := make([]interface{}, len(req.Params))
 	for i, p := range req.Params {
 		if p != nil {
-			err := json.Unmarshal([]byte(*p), &ethParams[i])
+			decodedParam := json.NewDecoder(bytes.NewReader([]byte(*p)))
+			decodedParam.UseNumber()
```

Comment:
From what I can read this stores the number in the string representation, have you checked that FireFly Signer supports this?

Reply:
Just thinking of how the RLP encode code there would treat this

Reply:
So it feels like it was just going wrong before it hit FireFly Signer https://github.com/hyperledger/firefly-signer/blob/5256116539ed83a23a52103f6260f4887abe0e85/pkg/abi/abiencode.go#L181

Reply:
I suspect we need to fix the issue in https://github.com/hyperledger/firefly-common/blob/main/pkg/fftypes/jsonany.go#L51

Reply:
Trying to understand this. The code before seemed correct:
```go
// Parse the params into the standard semantics of Go JSON unmarshalling, with []interface{}
```

Reply:
The intent of this code should be to take:
- take raw `json.RawMessage` etc still input from `req.Params`
- for each, do a standard `json.Unmarshal()` to `interface{}` so Golang picks the type
- pass that down to `decodedParam.Decode` to do type-specific processing on